embed/config.go

```diff
@@ -344,16 +328,25 @@ func (cfg Config) InitialClusterFromName(name string) (ret string) {
 func (cfg Config) IsNewCluster() bool { return cfg.ClusterState == ClusterStateFlagNew }
 func (cfg Config) ElectionTicks() int { return int(cfg.ElectionMs / cfg.TickMs) }
 
-// IsDefaultHost returns the default hostname, if used, and the error, if any,
-// from getting the machine's default host.
-func (cfg Config) IsDefaultHost() (string, error) {
-	if len(cfg.APUrls) == 1 && cfg.APUrls[0].String() == DefaultInitialAdvertisePeerURLs {
-		return defaultHostname, defaultHostStatus
+// IsDefaultHost returns true if listen URL is 0.0.0.0 and needs detect host name.
+func (cfg Config) IsDefaultHost() bool {
+	if len(cfg.LPUrls) == 1 {
+		ip, _, _ := net.SplitHostPort(cfg.LPUrls[0].Host)
+		if ip == "0.0.0.0" {
+			if len(cfg.APUrls) == 1 && cfg.APUrls[0].String() == DefaultInitialAdvertisePeerURLs {
+				return true
+			}
+		}
 	}
-	if len(cfg.ACUrls) == 1 && cfg.ACUrls[0].String() == DefaultAdvertiseClientURLs {
-		return defaultHostname, defaultHostStatus
+	if len(cfg.LCUrls) == 1 {
+		ip, _, _ := net.SplitHostPort(cfg.LCUrls[0].Host)
+		if ip == "0.0.0.0" {
+			if len(cfg.ACUrls) == 1 && cfg.ACUrls[0].String() == DefaultAdvertiseClientURLs {
+				return true
+			}
+		}
 	}
-	return "", defaultHostStatus
+	return false
```

Comment:
this changes the meaning of the function though? This function was meant to say whether it's using the default hosts /  user didn't specify, but now it's returning true if it's overriding the hosts.